Feature/populate footer

XDRO opened the pull request:
## Describe your changes
Create query to keystone server company contact info; pull information from server; implement returned information into footer; create window.onload function inside of main to refetch information on page reload. Conditionally render said information based on if values return truthy.
## Issue ticket number and link
EES-20
https://tripleten-apiary.atlassian.net/browse/EES-20
## Checklist before requesting a review

- [ ] I have performed a self-review of my code
- [ ] If there are changes to components, I have added / updated automated tests
- [ ] I have made any necessary updates to Storybook to accompany these changes
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings

Commit:
intial commit for populate footer branch, download apollo client, use query and gql, replace hardcoded values, change style for address

Commit:
update objects to to be more uniform when accessing information

Commit:
create window onload refetch function in main

Commit:
create file in graphql folder to seperate query from footer component

Commit:
create ternary operand to return contect information if contact information returns true and has an item iin the array return null otherwise

Commit:
Merge remote-tracking branch 'origin/main' into feature/populate-footer

XDRO committed:
change data 2022 back to new date get full year function

RPinkha commented on src/components/Footer/Footer.tsx:
```diff
@@ -23,39 +36,39 @@ export const Footer = ({ screenWidth = "desktop" }: Props) => {
+                className={`font-text text-paragraph-sm font-normal leading-[26px] tracking-[-0.02em] text-left sm:text-center cursor-pointer ${
+                  isMobile ? "text-center" : "text-left"
+                }`}
+                onClick={() => console.log("clicked email")}
```
Please compare with Main and see if this matches the footer design, because for the footer we removed all the console.log statements, and they are here now.

RPinkha commented on src/components/Footer/Footer.tsx:
```diff
@@ -1,14 +1,27 @@
+  }
+
+  const companyContactInfo = useCompanyContactInfo();
+  console.log(companyContactInfo);
```
Need to remove this console.log statement.

RPinkha reviewed (commented):
Please also pull from main as there are some merge conflicts that need resolving.

RPinkha committed:
fix: removed console log statements and added mailTo to be in line with main

RPinkha committed:
feat: pulled from main

RPinkha reviewed (approved):
Code looks good to me. 
Approved for merger on my end.

RPinkha commented on src/components/Footer/Footer.tsx:
```diff
@@ -23,39 +35,40 @@ export const Footer = ({ screenWidth = "desktop" }: Props) => {
+                }`}
+                onClick={() => {}}
+              >
+                {companyContactInfo ? companyContactInfo[0]?.address : ""}
```
The address seems to not render correctly for me now that it is connected to the backend. 
And that's because on the backend, the address field isn't asking for a full address. 
This can be mitigated by either asking for the full address from the client on the backend, or adding more field (like zip code and state) that are missing from the backend and then concatenating them to make the full address.

XDRO replied:
I am not sure how you're address renders, but when I run my backend this is what the footer looks like for me.
<img width="854" alt="Screenshot 2024-09-21 at 11 01 36" src="https://github.com/user-attachments/assets/10c697f7-c380-46c4-af05-2d2770d6a970">
Also I've attached a screenshot of how I have set up the data in keystone 
<img width="1230" alt="image" src="https://github.com/user-attachments/assets/4f3cb8f8-d8bf-40b6-bed8-9b00bbc09f64">

XDRO replied:
So I am not sure what yours looks like when you are running the backend and the frontend together, but this is what mine looks like when I run `npm run dev` on the backend and frontend together. 
<img width="1652" alt="image" src="https://github.com/user-attachments/assets/94b2e21c-bf2a-43cb-8a80-38b128c35ee4">
 Also here is what my backend data looks like: 
<img width="1652" alt="image" src="https://github.com/user-attachments/assets/3ea4d398-5f5e-48a0-ab32-2465ddc335bb">
 The issues you're encountering could have something to do with the way you have structured this list on your end.

RPinkha reviewed (commented):
The address renders incorrectly.

tomershvueli commented:
@XDRO I'm seeing an error in Storybook when trying to view the Footer component on this branch (latest code after `npm install`, and backend on latest code with a Company Contact defined): 
![image](https://github.com/user-attachments/assets/901496b3-db39-4e96-8d45-992379edd7a6)

What am I missing here?

Commit:
wrap footer component in stories with apollo provider; import apollo client, provider, and in memory cache; define client inside of stories file

XDRO commented:
> @XDRO I'm seeing an error in Storybook when trying to view the Footer component on this branch (latest code after `npm install`, and backend on latest code with a Company Contact defined): ![image](https://private-user-images.githubusercontent.com/1024355/369655906-901496b3-db39-4e96-8d45-992379edd7a6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjY5NjUyOTEsIm5iZiI6MTcyNjk2NDk5MSwicGF0aCI6Ii8xMDI0MzU1LzM2OTY1NTkwNi05MDE0OTZiMy1kYjM5LTRlOTYtOGQ0NS05OTIzNzllZGQ3YTYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkyMiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MjJUMDAyOTUxWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ODc3ZjhkZDAwMzQzMDc4ZWFjZmMwMTlmZGExMTBhNjI5ZGQwNWU5YzFmNjM4ZGY5Njg0NjMwYjkxM2U5MGU2NCZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.6fkvqNdOM2Y8eNQNgl_k2dK3lmownYKAtia3vDoHZ9k)


> 
> What am I missing here?

I've updated the stories file and I have fixed this issue. It was caused because the footer in stories needed to be wrapped inside of an <ApolloProvider> tag, and define the client inside of the stories file as well. This was because the Footer component relies on Apollo Client. I needed to provide an ApolloProvider to supply the Apollo Client context to the component. Here is a screenshot of the footter story working. <img width="2045" alt="Screenshot 2024-09-21 at 19 43 12" src="https://github.com/user-attachments/assets/99ba8380-2f72-40ef-b2a1-78dbfba2fb5c"> @tomershvueli

RPinkha committed:
feat: pulled changes from main

tomershvueli commented on src/components/Footer/Footer.tsx:
```diff
@@ -23,39 +35,40 @@ export const Footer = ({ screenWidth = "desktop" }: Props) => {
+                  isMobile ? "text-center" : "text-left"
+                }`}
+                onClick={() => {}}
+                href="mailto:Info@experteventsolutions.com"
```
The email link should also be dynamic and come from our `companyContactInfo`.

tomershvueli commented on src/components/Footer/Footer.tsx:
```diff
@@ -23,39 +35,40 @@ export const Footer = ({ screenWidth = "desktop" }: Props) => {
+            }`}
+          >
+            <div className="flex flex-col gap-[24px] ">
+              <p
```
nit: these components could also not be rendered at all if we don't have the specific piece of information that they're trying to render.

RPinkha committed:
feat: href mailto attribute now uses value from server

RPinkha committed:
fix: moved outdoors image to images and deleted assets folder'

RPinkha committed:
Merge branch 'main' into feature/populate-footer

RPinkha committed:
fix: fixed paths for moved image

RPinkha committed:
feat: added conditional rendering on the p and a tags

tomershvueli commented on src/components/Footer/Footer.tsx:
```diff
@@ -42,32 +42,36 @@ export const Footer = ({ screenWidth = "desktop" }: Props) => {
+                    isMobile ? "text-center" : "text-left"
+                  }`}
+                >
+                  {companyContactInfo[0]?.companyName}
```
nit: If we have the check above (L#45), we don't need the optional chaining here, since we know `companyContactInfo[0].companyName` exists.

tomershvueli reviewed (approved):
LGTM!
